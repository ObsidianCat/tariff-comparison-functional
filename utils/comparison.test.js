const compareProducts = require('./comparison').compareProducts
const prepareComparisonForPrint = require('./comparison').prepareComparisonForPrint

//use my mock data as a test mock
const tariffs = require('../data/tariffs').tariffsList
const mockInput = '4500'
it('should produce sorted array of comparisons', () => {
  const comparisonResults = compareProducts(mockInput, tariffs)
  expect(comparisonResults).toMatchSnapshot();
});

it('should produce a formatted and readable comparisons overview', () => {
  const comparisonResults = compareProducts(mockInput, tariffs)
  const presentationResults = prepareComparisonForPrint(comparisonResults)
  expect(presentationResults).toMatchSnapshot();
});