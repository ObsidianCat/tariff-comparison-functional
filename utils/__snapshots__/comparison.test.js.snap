// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should produce a formatted and readable comparisons overview 1`] = `
"Packaged tariff: 950.00 EUR 
basic electricity tariff: 1050.00 EUR 
"
`;

exports[`should produce sorted array of comparisons 1`] = `
Array [
  Object {
    "message": "Packaged tariff: 950.00 EUR",
    "precisePrice": 95000,
  },
  Object {
    "message": "basic electricity tariff: 1050.00 EUR",
    "precisePrice": 105000,
  },
]
`;
